api/prometheus/v1/api.go

```diff
@@ -477,10 +477,12 @@ func (c apiClient) Do(ctx context.Context, req *http.Request) (*http.Response, [
 
 	var result apiResponse
 
-	if err = json.Unmarshal(body, &result); err != nil {
-		return resp, body, &Error{
-			Type: ErrBadResponse,
-			Msg:  err.Error(),
+	if 204 != code {
```

Comment:
Maybe use `http.StatusNoContent` instead of plain `204` here.

Reply:
Good call. I've fixed that.